### DISCUSSION TITLE
Why do Groups change size when you add a shape to them?

### DISCUSSION THREAD

**Discussion by overdev-l** [fabricjs/fabric.js#10369] [Q&A]
I am using V6.5.3, I found that `addWithUpdate` is removed after group rewrite and`add` function has its behavior, I want to keep the size and position of the group unchanged when I add it to the group and I will set the crop on the group to achieve the effect of overflow hiding, what should I do?

**Comment by asturur** [comment]
There is a new tool for that, is called the layout manager.
Check for some example of the fixed layout manager in the tests to see how to use it.

**Reply by overdev-l** [reply]
I found that there is very little description of it in the latest documents. Where should I see its usage and performance? Is there a demo I can view?

**Reply by asturur** [reply]
Yes there is little description is true, and no demo.
Thise needs to be fixed.

I think what you want is something like this:

```ts
import { FixedLayout, LayoutManager, Group, } from 'fabric';
// create group with a layoutManager with fixed strategy
const grp = new Group([...objects], { layoutManager: new LayoutManager(new FixedLayout()) });
// if the above group does not have the correct size
const grp2 = new Group([...objects]);
grp2.layoutManager = new LayoutManager(new FixedLayout());
```

I do not rememember if there are differences by passing the layout manager in the constructor or setting it later

**Reply by overdev-l** [reply]
> There is a new tool for that, is called the layout manager.有一个新工具可以实现这一点，称为布局管理器。 Check for some example of the fixed layout manager in the tests to see how to use it.检查测试中固定布局管理器的一些示例以了解如何使用它。

Yes, I used this layout but had other problems, you can check the https://github.com/fabricjs/fabric.js/issues/10381#issuecomment-2572052030
